collect_app/src/main/java/org/odk/collect/android/application/initialization/AnalyticsInitializer.kt

```diff
@@ -0,0 +1,24 @@
+package org.odk.collect.android.application.initialization
+
+import org.odk.collect.analytics.Analytics
+import org.odk.collect.android.preferences.keys.GeneralKeys
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.version.VersionInformation
+
+class AnalyticsInitializer(
```

Comment:
As mentioned in the PR description, I've been eyeing up the App Startup library. A lot of app frameworks (across different platforms) use this multi object initialization style so moving to it felt pretty natural, allows for us to unit test individual chunks of init logic and should fit in well with App Startup when we do get there.